vpr/src/pack/cluster.cpp

```diff
@@ -2128,13 +2150,22 @@ static void update_cluster_stats(const t_pack_molecule* molecule,
         VTR_ASSERT(atom_pb);
 
         cur_pb = atom_pb->parent_pb;
+
+        //Update attraction group
+        AttractGroupId atom_grp_id = attraction_groups.get_atom_attraction_group(blk_id);
+
         while (cur_pb) {
             /* reset list of feasible blocks */
             if (cur_pb->is_root()) {
                 cb = cur_pb;
             }
             cur_pb->pb_stats->num_feasible_blocks = NOT_VALID;
             cur_pb->pb_stats->num_child_blocks_in_pb++;
+
+            if (atom_grp_id != AttractGroupId::INVALID()) {
```

Comment:
To-Do comment - allow more than one attraction group

Reply:
otherwise you are overwriting the attraction group that was previously in the cluster

Reply:
can have a vector of attraction groups for each cluster